File: mixin-base/src/main/java/net/yakclient/mixin/base/registry/pool/ClassLocation.java
package net.yakclient.mixin.base.registry.pool;

import java.util.Objects;

public class ClassLocation implements Location {
    private final String cls;

    public ClassLocation(String cls) {
        this.cls = cls;
    }

    public static ClassLocation fromDataDest(MixinMetaData data) {
        return new ClassLocation(data.getClassTo());
    }

    public static ClassLocation fromDataOrigin(MixinMetaData data) {
        return new ClassLocation(data.getClassFrom());
    }

    public String getCls() {
        return cls;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        final var that = (ClassLocation) o;
        return Objects.equals(cls, that.cls);
    }

    @Override
    public int hashCode() {
        return Objects.hash(cls);
    }
}

File: mixin-base/src/main/kotlin/net/yakclient/mixin/base/registry/pool/ClassLocation.kt
package net.yakclient.mixin.base.registry.pool

import java.util.*

open class ClassLocation(val cls: String) : Location {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ClassLocation
        return cls == that.cls
    }

    override fun hashCode(): Int {
        return Objects.hash(cls)
    }

    companion object {
        fun fromDataDest(data: MixinMetaData): ClassLocation {
            return ClassLocation(data.classTo)
        }

        fun fromDataOrigin(data: MixinMetaData): ClassLocation {
            return ClassLocation(data.classFrom)
        }
    }
}